models.py

```diff
@@ -15,17 +15,17 @@ def __str__(self):
     def shortened(self):
         return '%s/s/%s' % (settings.BASE_URL, self.short_id)
 
+    @classmethod
+    def create(cls, long_url):
```

Comment:
Would be consistent with Django to name this `get_or_create`, since that's what it's really doing.

Reply:
But also I'd be wary for putting this as a class method on the model object. Since then there's both `Url.objects.get_or_create()` and `Url.get_or_create`. Something to think about even if the name stays as `create`. I agree though that it shouldn't just be a "loose" function. Not sure if there's an established django pattern for this out there somewhere.

Reply:
I agree renaming to `get_or_create` makes sense. That's a good idea. As for where to put the method see [this](https://docs.djangoproject.com/en/2.2/ref/models/instances/#creating-objects). I would be easy to either, but it seems using a custom manager is the recommended way.